src/vivarium_public_health/metrics/disease.py

```diff
@@ -65,140 +36,62 @@ def __init__(self, disease: str):
-            }
-        }
-
-    ##############
-    # Properties #
-    ##############
-
     @property
     def name(self):
-        return f"disease_observer.{self.disease}"
-
-    #################
-    # Setup methods #
-    #################
+        return f"disease_observer_{self.disease}"
 
     # noinspection PyAttributeOutsideInit
     def setup(self, builder: Builder) -> None:
-        self.config = self._get_stratification_configuration(builder)
-        self.stratifier = self._get_stratifier(builder)
-        self.states = self._get_states(builder)
-        self.transitions = self._get_transitions(builder)
-        self.population_view = self._get_population_view(builder)
-
-        self.counts = Counter()
-
-        self._register_simulant_initializer(builder)
-        self._register_time_step_prepare_listener(builder)
-        self._register_collect_metrics_listener(builder)
-        self._register_metrics_modifier(builder)
-
-    def _get_stratification_configuration(self, builder: Builder) -> ConfigTree:
-        return builder.configuration.observers[self.disease]
-
-    # noinspection PyMethodMayBeStatic
-    def _get_stratifier(self, builder: Builder) -> ResultsStratifier:
-        return builder.components.get_component(ResultsStratifier.name)
-
-    def _get_states(self, builder: Builder) -> List[str]:
-        return builder.components.get_component(self.disease_component_name).state_names
-
-    def _get_transitions(self, builder: Builder) -> List[TransitionString]:
-        return builder.components.get_component(self.disease_component_name).transition_names
-
-    # noinspection PyMethodMayBeStatic
-    def _get_population_view(self, builder: Builder) -> PopulationView:
-        columns_required = [
-            self.current_state_column_name,
-            self.previous_state_column_name,
-        ]
-        return builder.population.get_view(columns_required)
-
-    def _register_simulant_initializer(self, builder: Builder) -> None:
-        # todo observer should not be modifying state table
-        builder.population.initializes_simulants(
-            self.on_initialize_simulants,
-            creates_columns=[self.previous_state_column_name],
-        )
+        self.step_size = builder.time.step_size()
+        self.config = builder.configuration.stratification[self.disease]
 
-    def _register_time_step_prepare_listener(self, builder: Builder) -> None:
-        # In order to get an accurate representation of person time we need to look at
-        # the state table before anything happens.
-        builder.event.register_listener("time_step__prepare", self.on_time_step_prepare)
-
-    def _register_collect_metrics_listener(self, builder: Builder) -> None:
-        builder.event.register_listener("collect_metrics", self.on_collect_metrics)
+        disease_model = builder.components.get_component(f"disease_model.{self.disease}")
+        self.previous_state_column = f"previous_{self.disease}"
```

Comment:
You are calling this twice - here and in the constructor.

Reply:
good eye! thanks!